// @wordpress/rich-text/build-module/index.js
import { store } from "./store";
import { applyFormat } from "./apply-format";
import { concat } from "./concat";
import { RichTextData, create } from "./create";
import { getActiveFormat } from "./get-active-format";
import { getActiveFormats } from "./get-active-formats";
import { getActiveObject } from "./get-active-object";
import { getTextContent } from "./get-text-content";
import { isCollapsed } from "./is-collapsed";
import { isEmpty } from "./is-empty";
import { join } from "./join";
import { registerFormatType } from "./register-format-type";
import { removeFormat } from "./remove-format";
import { remove } from "./remove";
import { replace } from "./replace";
import { insert } from "./insert";
import { insertObject } from "./insert-object";
import { slice } from "./slice";
import { split } from "./split";
import { toDom } from "./to-dom";
import { toHTMLString } from "./to-html-string";
import { toggleFormat } from "./toggle-format";
import { unregisterFormatType } from "./unregister-format-type";
import { createElement } from "./create-element";
import { useAnchorRef } from "./component/use-anchor-ref";
import { useAnchor } from "./component/use-anchor";
import {
  default as default2,
  useRichText
} from "./component";
export {
  RichTextData,
  default2 as __experimentalRichText,
  createElement as __unstableCreateElement,
  toDom as __unstableToDom,
  useRichText as __unstableUseRichText,
  applyFormat,
  concat,
  create,
  getActiveFormat,
  getActiveFormats,
  getActiveObject,
  getTextContent,
  insert,
  insertObject,
  isCollapsed,
  isEmpty,
  join,
  registerFormatType,
  remove,
  removeFormat,
  replace,
  slice,
  split,
  store,
  toHTMLString,
  toggleFormat,
  unregisterFormatType,
  useAnchor,
  useAnchorRef
};
