// @wordpress/components/build-module/navigator/legacy.js
import { Navigator as InternalNavigator } from "./navigator/component";
import { NavigatorScreen as InternalNavigatorScreen } from "./navigator-screen/component";
import { NavigatorButton as InternalNavigatorButton } from "./navigator-button/component";
import { NavigatorBackButton as InternalNavigatorBackButton } from "./navigator-back-button/component";
import { NavigatorToParentButton as InternalNavigatorToParentButton } from "./navigator-to-parent-button/component";
import { useNavigator } from "./use-navigator";
const NavigatorProvider = Object.assign(InternalNavigator, {
  displayName: "NavigatorProvider"
});
const NavigatorScreen = Object.assign(InternalNavigatorScreen, {
  displayName: "NavigatorScreen"
});
const NavigatorButton = Object.assign(InternalNavigatorButton, {
  displayName: "NavigatorButton"
});
const NavigatorBackButton = Object.assign(InternalNavigatorBackButton, {
  displayName: "NavigatorBackButton"
});
const NavigatorToParentButton = Object.assign(InternalNavigatorToParentButton, {
  displayName: "NavigatorToParentButton"
});
export {
  NavigatorBackButton,
  NavigatorButton,
  NavigatorProvider,
  NavigatorScreen,
  NavigatorToParentButton,
  useNavigator
};
