// @wordpress/components/build-module/mobile/keyboard-aware-flat-list/index.android.js
import { FlatList } from "react-native";
import Animated from "react-native-reanimated";
import { forwardRef, useCallback, useImperativeHandle } from "@wordpress/element";
import useScroll from "./use-scroll";
import KeyboardAvoidingView from "../keyboard-avoiding-view";
import { OPTIMIZATION_ITEMS_THRESHOLD, OPTIMIZATION_PROPS } from "./shared";
import { jsx as _jsx } from "react/jsx-runtime";
const AnimatedFlatList = Animated.createAnimatedComponent(FlatList);
const EMPTY_OBJECT = {};
const KeyboardAwareFlatList = ({
  onScroll,
  ...props
}, ref) => {
  const {
    extraScrollHeight,
    scrollEnabled,
    shouldPreventAutomaticScroll
  } = props;
  const {
    scrollViewRef,
    scrollHandler,
    scrollToSection,
    scrollToElement,
    onContentSizeChange,
    lastScrollTo
  } = useScroll({
    scrollEnabled,
    shouldPreventAutomaticScroll,
    extraScrollHeight,
    onScroll
  });
  const getFlatListRef = useCallback((flatListRef) => {
    scrollViewRef.current = flatListRef?.getNativeScrollRef();
  }, [scrollViewRef]);
  useImperativeHandle(ref, () => {
    return {
      scrollViewRef: scrollViewRef.current,
      scrollToSection,
      scrollToElement,
      lastScrollTo
    };
  });
  const optimizationProps = props.data?.length > OPTIMIZATION_ITEMS_THRESHOLD ? OPTIMIZATION_PROPS : EMPTY_OBJECT;
  return /* @__PURE__ */ _jsx(KeyboardAvoidingView, {
    style: {
      flex: 1
    },
    children: /* @__PURE__ */ _jsx(AnimatedFlatList, {
      ref: getFlatListRef,
      onScroll: scrollHandler,
      onContentSizeChange,
      removeClippedSubviews: false,
      ...optimizationProps,
      ...props
    })
  });
};
var index_android_default = forwardRef(KeyboardAwareFlatList);
export {
  KeyboardAwareFlatList,
  index_android_default as default
};
